src/components/Dropdown/DropdownWithIcon.jsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { settings } from '../../constants/Settings';
+
+import { Dropdown } from './index';
+
+const { iotPrefix } = settings;
+
+const propTypes = {
+  itemToString: PropTypes.func,
+  items: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: PropTypes.string.isRequired,
+      text: PropTypes.string.isRequired,
+      icon: PropTypes.oneOfType([PropTypes.func, PropTypes.object]).isRequired,
+    })
+  ).isRequired,
+};
+
+const defaultPropTypes = {
+  itemToString: null,
+};
+
+const DropdownWithIcon = ({
+  selectedViewId,
+  items,
+  itemToString,
+  actions: { onChangeView, ...otherActions },
+  ...other
+}) => {
+  const onSelectionChange = (changes) => {
+    const { selectedItem } = changes;
+
+    onChangeView(selectedItem);
+  };
+
+  const renderLabel = (item) => {
+    return (
+      <>
+        <div className={`${iotPrefix}--dropdown__label`}>
+          {item?.icon !== undefined ? React.createElement(item?.icon) : null}
+          <div className={`${iotPrefix}--dropdown__label__content`}>
+            {item ? item.text : ''}
+          </div>
+        </div>
+      </>
+    );
+  };
+
+  return (
+    <Dropdown
+      items={items}
+      actions={otherActions}
+      onChange={onSelectionChange}
+      {...other}
+      itemToString={itemToString !== null ? itemToString : renderLabel}
+    />
+  );
+};
+
+DropdownWithIcon.propTypes = Dropdown.propTypes && propTypes;
```

Comment:
I don't understand this. You are just assigning `propTypes` but only if `Dropdown.propTypes` is truthy....

Reply:
I may need some direction for this, I want to merge the proptypes, but Dropdown's proptypes are not exported, should I copy them over to our index file from carbon and then append the Dropdown as dropdownProps and pass them along to the dropdown directly? Currently changing to an array of the props, it passes the validation as I want, but syntactically it looks wrong.

Reply:
Relooked into this, the second approach while passes validation, fails pr snapshots. The above passes validation for propTypes and passes snapshot tests, but I agree it is likely still incorrect. Any help on this would be appreciated.

Reply:
Since what you actually want is merging the two propTypes we need to either have the Dropdown's proptypes exported or copied. Perhaps @davidicus can weigh in